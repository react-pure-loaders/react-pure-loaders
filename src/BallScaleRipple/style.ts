import { keyframes } from '@emotion/react';
import { globalAnimation } from '../mixins';
import { PRIMARY_COLOR } from '../variables';

const ballScaleRipple = keyframes({
  '0%': {
    opacity: 1,
    transform: 'scale(0.1)',
  },
  '100%': {
    opacity: 0.0,
  },
  '70%': {
    opacity: 0.7,
    transform: 'scale(1)',
  },
});

export default (color?: string) => ({
  ' > div': {
    ...globalAnimation(),
    animation: `${ballScaleRipple} 1s 0s infinite cubic-bezier(.21,.53,.56,.8)`,
    border: `2px solid ${color || PRIMARY_COLOR}`,
    borderRadius: '100%',
    height: '50px',
    width: '50px',
  },
});
