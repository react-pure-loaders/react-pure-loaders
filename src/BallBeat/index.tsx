/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallBeat({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
