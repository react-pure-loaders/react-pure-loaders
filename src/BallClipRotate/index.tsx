/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallClipRotate({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)} data-testid="ball-clip-rotate-container">
                <div/>
            </div>
    );
  }

  return null;
}
