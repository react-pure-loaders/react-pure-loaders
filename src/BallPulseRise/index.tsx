/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallPulseRise({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
                <div/>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
