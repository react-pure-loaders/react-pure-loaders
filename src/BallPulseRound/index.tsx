/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallPulseRound({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
            </div>
    );
  }

  return null;
}
