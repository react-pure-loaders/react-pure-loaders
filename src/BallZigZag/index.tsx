/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallZigZag({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
