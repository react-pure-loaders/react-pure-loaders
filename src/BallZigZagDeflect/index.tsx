/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallZigZagDeflect({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
