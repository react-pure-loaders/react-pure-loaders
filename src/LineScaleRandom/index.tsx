/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
export default function LineScaleRandom({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div />
                <div />
                <div />
                <div />
            </div>
    );
  }

  return null;
}
