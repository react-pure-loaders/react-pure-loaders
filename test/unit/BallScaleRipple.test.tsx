import React from 'react';
import Chance from 'chance';
import { matchers } from '@emotion/jest';
import { render, cleanup } from '@testing-library/react';


import BallScaleRipple from '../../src/BallScaleRipple';
import { PRIMARY_COLOR } from '../../src/variables';

expect.extend(matchers);

const chance = new Chance();

describe('<BallScaleRipple>', () => {
    afterEach(cleanup);

    test('BallScaleRipple should match snapshot', () => {
        const { container } = render(<BallScaleRipple loading={true}/>);

        expect(container.firstChild).toMatchSnapshot();
    });

    test('BallScaleRipple should have default color', () => {
        const { container } = render(<BallScaleRipple loading={true}/>);

        expect(container.firstChild).toHaveStyleRule('border', `2px solid ${PRIMARY_COLOR}`, { target: '> div' });
    });

    test('BallScaleRipple should have given color', () => {
        const color = chance.color({ format: 'hex' });
        const { container } = render(<BallScaleRipple color={color} loading={true}/>);

        expect(container.firstChild).toHaveStyleRule('border', `2px solid ${color}`, { target: '> div' });
    });

    test('BallScaleRipple should have no children', () => {
        const { container } = render(<BallScaleRipple loading={false}/>);

        expect(container.firstChild).toBeNull();
    });

    test('BallScaleRipple should have three children', () => {
        const { container } = render(<BallScaleRipple loading={true}/>);

        expect(container.querySelectorAll('div')).toHaveLength(2);
    });
});
