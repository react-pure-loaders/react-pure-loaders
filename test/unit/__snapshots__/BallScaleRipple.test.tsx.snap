// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BallScaleRipple> BallScaleRipple should match snapshot 1`] = `
@keyframes animation-0 {
  0% {
    opacity: 1;
    -webkit-transform: scale(0.1);
    -moz-transform: scale(0.1);
    -ms-transform: scale(0.1);
    transform: scale(0.1);
  }

  100% {
    opacity: 0;
  }

  70% {
    opacity: 0.7;
    -webkit-transform: scale(1);
    -moz-transform: scale(1);
    -ms-transform: scale(1);
    transform: scale(1);
  }
}

.emotion-0>div {
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
  -webkit-animation: animation-0 1s 0s infinite cubic-bezier(.21,.53,.56,.8);
  animation: animation-0 1s 0s infinite cubic-bezier(.21,.53,.56,.8);
  border: 2px solid #ffffff;
  border-radius: 100%;
  height: 50px;
  width: 50px;
}

<div
  class="emotion-0"
>
  <div />
</div>
`;
